docs/self-managed/deployment/helm/configure/authentication-and-authorization/generic-oidc-provider.md

```diff
@@ -1,146 +1,784 @@
+  ...
+}
+```
+
+**Record these values for Helm configuration:**
+
+- `issuer` → Used for: `issuer`, `publicIssuerUrl`
+- `token_endpoint` → Used for: `tokenUrl`
+- `jwks_uri` → Used for: `jwksUrl`
+
+### Identify token claims
+
+Camunda needs to know which claims in access tokens identify users and clients. Claim names vary by provider.
+
+**Step 1: Obtain a test token**
+
+Use your OIDC client credentials to request an access token:
+
+```bash
+curl -X POST 'https://your-provider.example.com/oauth/token' \
+  -H 'Content-Type: application/x-www-form-urlencoded' \
+  -d 'client_id=<your-client-id>' \
+  -d 'client_secret=<your-client-secret>' \
+  -d 'grant_type=client_credentials'
+```
+
+This returns a JSON response containing an `access_token`.
+
+**Step 2: Decode the token**
+
+Decode the JWT to inspect its claims:
+
+```bash
+echo "<access-token>" | cut -d'.' -f2 | base64 -d | jq
+```
+
+Or use an online tool like [jwt.io](https://jwt.io) (paste the token to decode it).
+
+**Step 3: Identify required claims**
+
+Look for these claims in the decoded token:
+
+1. **User identification claim** (for web login)
+   - Identifies users uniquely
+   - Common names: `email`, `preferred_username`, `sub`, `upn`, `unique_name`
+   - You'll configure this as: `usernameClaim`
+
+2. **Client identification claim** (for machine-to-machine authentication)
+   - Identifies the calling client application
+   - Common names: `client_id`, `azp`, `appid`, `clientId`
+   - You'll configure this as: `clientIdClaim`
+
+3. **Audience claim** (`aud`)
+   - Specifies the intended audience for the token
+   - Often contains the client ID or a custom value
+   - You'll configure this as: `audience`
+
+**Common claim patterns by provider:**
+
+| Provider        | User Claim                      | Client Claim         | Audience Default       |
+| --------------- | ------------------------------- | -------------------- | ---------------------- |
+| Microsoft Entra | `preferred_username`            | `azp`                | Client ID              |
+| Keycloak        | `email` or `preferred_username` | `azp` or `client_id` | May need configuration |
+| Auth0           | `email`                         | `client_id`          | Client ID              |
+| Okta            | `email`                         | `client_id`          | Client ID              |
+
+:::tip Standard OIDC claims
+The OpenID Connect specification defines standard claims:
+
+- **User identification**: `sub` (subject) is always present and uniquely identifies the user. Providers may also include `preferred_username`, `email`, or other profile claims.
+- **Client identification**: `azp` (authorized party) or `client_id` depending on the provider's implementation. Both are standard OIDC claims.
+
+**Camunda's default claim mappings** (used if not explicitly configured):
+
+- `usernameClaim`: `preferred_username` (fallback to `sub` if not present)
+- `clientIdClaim`: `client_id`
```

Comment:
I think we don't have this as a unified concept across all components.

In the orchestration cluster, you can configure username claim and client id claim accordingly (and the defaults are `sub` and `<not set>`, i.e. all accesses are treated as users). In Web Modeler, the user id is always the `sub` claim and then it has a freely configurable (I think?!) user display name. And I'm not sure right now if it even has a concept for a client id. Didn't check the other components now.

In any case, I'm also not sure what this box here helps people with. Why did you include it? Maybe then we can find another way to formulate that.